From feee164dc085d0382998993f8e32fb9d50ac9363 Mon Sep 17 00:00:00 2001
From: "Dawei.Chien" <dawei.chien@mediatek.com>
Date: Mon, 22 Jun 2015 20:33:15 +0800
Subject: [PATCH 5/6] WIP: thermal: add device tree for MTK thermal cooling
 device.

Signed-off-by: Dawei Chien <dawei.chien@mediatek.com>

BUG=chrome-os-partner:40082
TEST=build and boot oak-rev1

Change-Id: I51cec23caec00777b7603d5b7e12d2ce166efe47
---
 arch/arm64/boot/dts/mediatek/mt8173.dtsi | 89 ++++++++++++++++++++++++++++++--
 1 file changed, 84 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8173.dtsi b/arch/arm64/boot/dts/mediatek/mt8173.dtsi
index c0649d0..e6aec07 100644
--- a/arch/arm64/boot/dts/mediatek/mt8173.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8173.dtsi
@@ -17,6 +17,7 @@
 #include <dt-bindings/reset-controller/mt8173-resets.h>
 #include <dt-bindings/power/mt8173-power.h>
 #include "mt8173-pinfunc.h"
+#include <dt-bindings/thermal/thermal.h>
 
 / {
 	compatible = "mediatek,mt8173";
@@ -54,6 +55,7 @@
 			reg = <0x000>;
 			enable-method = "psci";
 			cpu-idle-states = <&CPU_SLEEP_0>;
+			/* cooling options */
 			clocks = <&infracfg CLK_INFRA_CA53SEL>,
 				 <&apmixedsys CLK_APMIXED_MAINPLL>;
 			clock-names = "cpu", "intermediate";
@@ -102,6 +104,7 @@
 			reg = <0x100>;
 			enable-method = "psci";
 			cpu-idle-states = <&CPU_SLEEP_0>;
+			/* cooling options */
 			clocks = <&infracfg CLK_INFRA_CA57SEL>,
 				 <&apmixedsys CLK_APMIXED_MAINPLL>;
 			clock-names = "cpu", "intermediate";
@@ -181,21 +184,97 @@
 	};
 
         thermal-zones {
-                cpu_thermal: cpu_thermal {
+                cpu_thermal0: cpu_thermal0 {
                         polling-delay-passive = <1000>; /* milliseconds */
                         polling-delay = <0>; /* milliseconds */
 
                         thermal-sensors = <&thermal 0>;
 
                         trips {
-                                cpu_passive: cpu_passive {
-                                        temperature = <47000>; /* millicelsius */
+                                cpu_passive0: cpu_passive0 {
+                                        temperature = <68000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "passive";
+                                };
+                                cpu_crit0 {
+                                        temperature = <100000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "critical";
+                                };
+                        };
+
+			cooling-maps {
+				map0 {
+					trip = <&cpu_passive0>;
+					cooling-device = <&cpu0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+                };
+
+                cpu_thermal1: cpu_thermal1 {
+                        polling-delay-passive = <1000>; /* milliseconds */
+                        polling-delay = <0>; /* milliseconds */
+
+                        thermal-sensors = <&thermal 1>;
+
+                        trips {
+                                cpu_passive1: cpu_passive1 {
+                                        temperature = <68000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "passive";
+                                };
+
+                                cpu_crit1 {
+                                        temperature = <100000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "critical";
+                                };
+                        };
+                };
+
+                cpu_thermal2: cpu_thermal2 {
+                        polling-delay-passive = <1000>; /* milliseconds */
+                        polling-delay = <0>; /* milliseconds */
+
+                        thermal-sensors = <&thermal 2>;
+
+                        trips {
+                                cpu_passive2: cpu_passive2 {
+                                        temperature = <68000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "passive";
+                                };
+
+                                cpu_crit2 {
+                                        temperature = <100000>; /* millicelsius */
+                                        hysteresis = <2000>; /* millicelsius */
+                                        type = "critical";
+                                };
+			};
+
+                        cooling-maps {
+                                map2 {
+                                        trip = <&cpu_passive2>;
+                                        cooling-device = <&cpu2 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+                                };
+                        };
+                };
+
+                cpu_thermal3: cpu_thermal3 {
+                        polling-delay-passive = <1000>; /* milliseconds */
+                        polling-delay = <0>; /* milliseconds */
+
+                        thermal-sensors = <&thermal 3>;
+
+                        trips {
+                                cpu_passive3: cpu_passive3 {
+                                        temperature = <68000>; /* millicelsius */
                                         hysteresis = <2000>; /* millicelsius */
                                         type = "passive";
                                 };
 
-                                cpu_crit {
-                                        temperature = <90000>; /* millicelsius */
+                                cpu_crit3 {
+                                        temperature = <100000>; /* millicelsius */
                                         hysteresis = <2000>; /* millicelsius */
                                         type = "critical";
                                 };
-- 
1.9.1

